src/components/ReportingConfig/ReportingConfigForm.jsx

```diff
@@ -0,0 +1,512 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import isEmpty from 'lodash/isEmpty';
+import { ValidationFormGroup, Input, StatefulButton, Icon, Button } from '@edx/paragon';
+
+//  All the fields in this form that need to be validated can be added here.
+const REQUIRED_FIELDS = [
+  'hourOfDay',
+];
+// for the email delivery method
+const REQUIRED_EMAIL_FIELDS = [
+  ...REQUIRED_FIELDS,
+  'emailRaw',
+];
+// for the sftp delivery mothod
+const REQUIRED_SFTP_FIELDS = [
+  ...REQUIRED_FIELDS,
+  'sftpPort',
+  'sftpHostname',
+  'sftpUsername',
+  'sftpPassword',
+  'sftpFilePath',
+];
+
+// for the submit button
+const SUBMIT_STATES = {
+  ERROR: 'error',
+  DEFAULT: 'default',
+  COMPLETE: 'complete',
+  PENDING: 'pending',
+};
+
+const MONTHLY_MAX = 31;
+const MONTHLY_MIN = 1;
+
+const EMAIL_REGEX = /^[a-zA-Z0-9.!#$%&'*+/=?^_`{|}~-]+@[A-Z0-9.-]+\.[A-Z]{2,}$/igm;
```

Comment:
super nit: If we're doing case insensitive (`i`), we can drop the lower case letters from the beginning (`a-z`).

Reply:
If we already have a utility function (per Adam), that's probably a better solution.

Reply:
I've switched to the better utility function (Thanks Adam!)